### ISSUE TITLE
Theme Showcase: Various button contents are poorly aligned.

### ISSUE THREAD

**Issue by kwight** [Automattic/wp-calypso#55140]
The "Customize", "Info", and "Filters" buttons have at some point regressed, and now have their text and icons out of alignment.

<img width="1136" alt="Screen Shot 2021-08-03 at 2 23 02 PM" src="https://user-images.githubusercontent.com/349751/128088730-27b670b5-c7ea-4c5c-8d00-ffd0eee6a2da.png">

It appears a hard-coded height of 36px for `.components-button` may be causing the problem, contributed to by a `top: 0` for `.current-theme__post-revamp .current-theme__button .gridicon` that is maybe no longer needed? I'm not sure which height is supposed to be the correct one in this case.

**Comment by mreishus** [comment]
This looks to be caused by b95f4bdeb776acf3cfa299824924e2d56fa20bcd / #53221. I am unsure of the mechanism, but I see the misalignment happening on that commit, where the directly previous commit looks aligned 38f65341ad.

![2021-08-03_17-02](https://user-images.githubusercontent.com/937354/128092274-23da9ba1-8585-497c-a3ee-dbbd9381a2de.png)
![2021-08-03_17-01](https://user-images.githubusercontent.com/937354/128092281-5b847cf4-5806-4d35-9162-0b56d9c990c5.png)

**Comment by mreishus** [comment]
* Both Before/After: The `<a>` component has both `button` and `components-button` css classes. 
* Before: The `components-button` class, which has `display: inline-flex` "wins" and the button looks aligned.
* After: The `button` class, which has `display: inline-block` "wins" and the button looks misaligned.

https://user-images.githubusercontent.com/937354/128093028-3faeee73-ff10-4443-8799-3098c553d663.mp4

I do not know what about the patch causes the behavior change.

**Comment by mreishus** [comment]
I'm wondering if it's because the two rules are at the same level of specificity, there's no stable way to solve conflicts, so it uses the order the rules appeared, which  b95f4bdeb776acf flipped for some reason. At least that's what I'm seeing in my browser tools:

![2021-08-03_17-19](https://user-images.githubusercontent.com/937354/128093762-a7b3c94e-eed2-4a48-b5f1-83b699034ab8.png)
![2021-08-03_17-19_1](https://user-images.githubusercontent.com/937354/128093768-3d1f3a60-f921-4983-9781-e6b0742e9998.png)

If this is the case, then there might be more instance of this problem happening outside the theme showcase.

**Comment by Aurorum** [comment]
Just noting that I opened a PR yesterday in #55124 to fix those issues with specificity, but that won't help if this is a wider issue with buttons outside the Themes Showcase. However, I can't seem to find an example of this happening outside the Themes Showcase? For example, the buttons with Gridicons in the Media section seem unaffected.
